### ISSUE TITLE
Unable to deploy CommaFeed on OpenShift

### ISSUE THREAD

**Issue by icaroperseo** [Athou/commafeed#712]
I'm sorry to have to open [this thread](https://github.com/Athou/commafeed/issues/700) again but with your instructions I get the following error:

```
remote: [INFO] Including org.postgresql:postgresql:jar:9.4-1200-jdbc41 in the shaded jar.
remote: [INFO] Including com.github.dblock.waffle:waffle-jna:jar:1.7 in the shaded jar.
remote: [INFO] Including net.java.dev.jna:jna:jar:4.1.0 in the shaded jar.
remote: [INFO] Including net.java.dev.jna:jna-platform:jar:4.1.0 in the shaded jar.
remote: [INFO] Including net.sourceforge.jtds:jtds:jar:1.3.1 in the shaded jar.
Connection to commafeed-site.rhcloud.com closed by remote host.
fatal: The remote end hung up unexpectedly
error: error in sideband demultiplexer
To ssh://54e4111ce.......db000006@commafeed-site.rhcloud.com/~/git/commafeed.git/
   4cb5585..6d91216  master -> master
error: failed to push some refs to 'ssh://54e4111ce....db000006@commafeed-site.rhcloud.com/~/git/commafeed.git/'
```

Thanks in advance!

**Comment by tvvocold** [comment]
@Athou Me either, here is mine log
```
remote: [INFO] Changes detected - recompiling the module!
remote: [INFO] Compiling 122 source files to /var/lib/openshift/550141d6fcf933ec
bb000126/app-root/runtime/repo/target/classes
Connection to commafeed-domain.rhcloud.com closed by remote host.
fatal: The remote end hung up unexpectedly
error: error in sideband demultiplexer
To ssh://550141d6fcf933ecbb000126@commafeed-domain.rhcloud.com/~/git/commafeed.gi
t/
   f3e44ca..150cf1c  master -> master
error: failed to push some refs to 'ssh://550141d6fcf933ecbb000126@commafeed-domain.rhcloud.com/~/git/commafeed.git/'
```

**Comment by tvvocold** [comment]
@Athou I think maybe the memory reason? (only 500 MB memory & 1GB storage for one gear)

**Comment by Athou** [comment]
Looks like a timeout occuring when the build takes too long, seems to be an openshift issue. I'll check if there's something I can do.

**Comment by lxitgto** [comment]
I got the same problem. I try build it on my macbook and it will hold on "[INFO] --- frontend-maven-plugin:0.0.22:npm (npm install) @ commafeed ---
[INFO] Running 'npm install --color=false' in" long time.

**Comment by tvvocold** [comment]
@Athou Maybe it run out of gear?

**Comment by dgonzals** [comment]
Same thing happened to me. Then I tried it again and it said everything is updated. Still not working.

**Comment by Athou** [comment]
I don't have any workaround, really seems to be a timeout during the build.

**Comment by dgonzals** [comment]
Bummer. I don't know how to do it the other way.

**Comment by tvvocold** [comment]
@Athou 
Maybe it's not a time out , i think maybe it's OOM. I had try `git push --force`[1] still not working.

[1] https://github.com/boekkooi/openshift-diy-nginx-php/blob/e2b3c541de06e4634ef9ddac5d28fe614110d368/README.md#openshift-disconnects-on-build

**Comment by Athou** [comment]
They probably changed something on Openshift because I'm now able to build on a small gear without any issues by following the instructions in the readme.

**Comment by tvvocold** [comment]
Didn't work on mine :)
```
remote: [INFO] Downloading NPM from http://registry.npmjs.org/npm/-/npm-2.10.1.tgz to /var/lib/openshift/55657f3f5973ca718e000155/app-root/runtime/repo/npm.tar.
gz
remote: [INFO] No proxy was configured, downloading directly
remote: [INFO] Extracting NPM files in node/
remote: [INFO] Unpacking /var/lib/openshift/55657f3f5973ca718e000155/app-root/runtime/repo/npm.tar.gz into /var/lib/openshift/55657f3f5973ca718e000155/app-root/
runtime/repo/node
remote: [INFO] Installed NPM locally.
remote: [INFO]
remote: [INFO] --- frontend-maven-plugin:0.0.22:npm (npm install) @ commafeed ---
remote: [INFO] Running 'npm install --color=false' in /var/lib/openshift/55657f3f5973ca718e000155/app-root/runtime/repo
Connection to commafeed-ttvvoo.rhcloud.com closed by remote host.
fatal: The remote end hung up unexpectedly
error: error in sideband demultiplexer
To ssh://55657f3f5973ca718e000155@commafeed-ttvvoo.rhcloud.com/~/git/commafeed.git/
   c3291ef..50355e6  master -> master
error: failed to push some refs to 'ssh://55657f3f5973ca718e000155@commafeed-ttvvoo.rhcloud.com/~/git/commafeed.git/'
```

**Comment by srirams** [comment]
I'm pretty sure the problem is running out of memory space on a small gear (and the gear restarts, causing the disconnect). Since the problem seems to be in compiling, and not in running, is there any way to get a compiled version that can run on a small gear?

**Comment by srirams** [comment]
ok, I've got it working (I can finally update from 2.0.3 which I've been running for a long time).

Just create a new gear in openshift, and add the diy and mysql cartridges.

use this as the deploy hook (.openshift/action_hooks/deploy):

```
#!/bin/bash
cd $OPENSHIFT_REPO_DIR
sed -i 's/@OPENSHIFT_DIY_IP@/'"$OPENSHIFT_DIY_IP"'/g' .openshift/config.mysql.yml
sed -i 's/@OPENSHIFT_DIY_PORT@/'"$OPENSHIFT_DIY_PORT"'/g' .openshift/config.mysql.yml
sed -i 's/@OPENSHIFT_APP_DNS@/'"$OPENSHIFT_APP_DNS"'/g' .openshift/config.mysql.yml
sed -i 's/@OPENSHIFT_APP_NAME@/'"$OPENSHIFT_APP_NAME"'/g' .openshift/config.mysql.yml
sed -i 's/@OPENSHIFT_MYSQL_DB_HOST@/'"$OPENSHIFT_MYSQL_DB_HOST"'/g' .openshift/config.mysql.yml
sed -i 's/@OPENSHIFT_MYSQL_DB_USERNAME@/'"$OPENSHIFT_MYSQL_DB_USERNAME"'/g' .openshift/config.mysql.yml
sed -i 's/@OPENSHIFT_MYSQL_DB_PASSWORD@/'"$OPENSHIFT_MYSQL_DB_PASSWORD"'/g' .openshift/config.mysql.yml
cd $OPENSHIFT_DATA_DIR
wget https://github.com/Athou/commafeed/releases/download/2.2.0/commafeed.jar
```

make sure you make it executable in git.

use this as the start hook (.openshift/action_hooks/start):

```
#!/bin/bash
# The logic to start up your application should be put in this
# script. The application will work only if it binds to
# $OPENSHIFT_DIY_IP:8080
cd $OPENSHIFT_DATA_DIR
export JAVA_HOME=/etc/alternatives/java_sdk_1.8.0
nohup $JAVA_HOME/bin/java -jar commafeed.jar server $OPENSHIFT_REPO_DIR/.openshift/config.mysql.yml > ${OPENSHIFT_DIY_LOG_DIR}/commafeed.log 2>&1 &
```

use the same stop script that is the current commafeed repo.

And that's it! You don't need to clone the source or build or do anything else.

**Comment by Poorchop** [comment]
@srirams, your instructions worked perfectly with 2.3.0 (latest release as of typing this). However, I'm unable to log in. I copied .openshift/config.mysql.yml and according to the file, the database username and password should be the proper credentials. That doesn't seem to be the case for me though. Am I missing something?

**Comment by srirams** [comment]
I'm assuming you tried the default admin/admin?

**Comment by Poorchop** [comment]
I didn't realize that was the default login. I missed this in the README:

>The default user is admin and the default password is admin.

It's all working now. Hopefully I'll be able to add some feeds and finally start using this! Thanks for your help and for taking the time to type out the instructions. I'm not sure how upgrading will work but hopefully it doesn't involve deleting the entire gear and starting over from scratch.

**Comment by srirams** [comment]
Glad you got it working :)

Upgrading should be as simple as stopping the gear and replacing the old commafeed.jar with the new one, and then starting the gear back up. Everything's been very stable for me (for reference, I haven't had to touch a thing since my last post in August, and I use Commafeed quite heavily)

**Comment by Poorchop** [comment]
I updated the config manually since I couldn't seem to save my changes via the web interface and I noticed that it downloaded commafeed.jar again, so you're right, updating should be as simple as replacing the link to the latest release. Now I just have to figure out how to delete the unused .jar, so I'll start going through their docs.

Maybe your instructions can be added to the wiki? I almost gave up until I luckily saw your steps after some digging around.
